### PR TITLE
[202305] Enhancement for route perf test for backend devices

### PR DESCRIPTION
**Pull request by developfast**
### Description of PR

Summary: cherry-pick pr for 202305 from #12491 
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [X] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311

### CODE REVIEW

**Review by neethajohn — APPROVED** [review]

**Comment by developfast** [comment]
@StormLiangMS can you pls help merge?

File: tests/route/test_route_perf.py [hunk 1]
```diff
@@ -413,9 +425,9 @@ def test_perf_add_remove_routes(
         )
 
         # Traffic verification with 10 random routes
-        mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
```
Comment (StormLiangMS) [inline tests/route/test_route_perf.py:416]: what's the reason to remove the mg_facts here?

File: tests/route/test_route_perf.py [hunk 2]
```diff
@@ -413,9 +425,9 @@ def test_perf_add_remove_routes(
...
         port_indices = mg_facts["minigraph_ptf_indices"]
-        nexthop_intf = str_intf_nexthop["ifname"].split(",")
+        # split off the vlan id from the interface name separated by the . delimiter
+        nexthop_intf = [nh_intf.split(".")[0] for nh_intf in str_intf_nexthop["ifname"].split(",")]
```
Comment (StormLiangMS) [inline tests/route/test_route_perf.py:430]: this change seems not only for backend, could you also run test with other topo device to avoid regression?

**Comment by neethajohn** [comment]
Tested with 202305 on backend and non backend testbed and tests have passed

image


image

**Review by StormLiangMS — APPROVED** [review]
LGTM

**Pull request merged**